### DISCUSSION TITLE
how to ignore custom block `docs` in sfc?

### DISCUSSION THREAD

**Discussion by rainmanhhh** [posva/unplugin-vue-router#420] [Questions and Help] [answered]
[Vue Router warn]: Unexpected error when starting the router: 
SyntaxError: The requested module '/src/components/page/PageTop.vue?vue&type=docs&index=0&lang.docs' does not provide an export named 'default'

I got above error after adding custom block docs in sfc component PageTop.vue(it's a navigation bar which will be used in route page):
```
<docs>
this is navigation bar
</docs>

<script lang="ts" setup>
//...
</script>
```

**Comment by posva** [comment]
À boiled down repro would help 🙂

**Comment by rainmanhhh** [comment]
> À boiled down repro would help

https://github.com/rainmanhhh/vitesse-lite-debug

forked from vitesse-lite, just add a `docs` block at src/pages/index.vue

**Comment by posva** [comment, marked as answer]
Thanks! The issue seems to be because the generated component by the vue vite plugin does this:

![Screenshot 2024-06-13 at 10 10 24](https://github.com/posva/unplugin-vue-router/assets/664177/3cdc424e-ff1b-4e92-ba8a-55d0840e0f27)

The router fails then.

My guess is that you need something to handle that custom block but this is not related to this plugin despite the error being reported by the router

**Reply by rainmanhhh** [reply]
then what is the "official" way to tell vite ignore the custom block `docs`? I think `route` is also a custom block introduced by unplugin-vue-router
